zebrad/tests/acceptance.rs

```diff
@@ -1473,3 +1475,81 @@ where
 
     Ok(())
 }
+
+#[tokio::test]
+#[ignore]
+async fn fully_synced_rpc_test() -> Result<()> {
+    zebra_test::init();
+
+    // TODO: reuse code from https://github.com/ZcashFoundation/zebra/pull/4177/
+    // to get the cached_state_path
+    const CACHED_STATE_PATH_VAR: &str = "ZEBRA_CACHED_STATE_PATH";
+    let cached_state_path = match env::var_os(CACHED_STATE_PATH_VAR) {
+        Some(argument) => PathBuf::from(argument),
+        None => {
+            tracing::info!(
+                "skipped send transactions using lightwalletd test, \
+                 set the {CACHED_STATE_PATH_VAR:?} environment variable to run the test",
+            );
+            return Ok(());
+        }
+    };
+
+    let network = Network::Mainnet;
+
+    let (_zebrad, zebra_rpc_address) =
+        spawn_zebrad_for_rpc_without_initial_peers(network, cached_state_path)?;
+
+    // Make a getblock test that works only on synced node (high block number).
+    // The block is before the mandatory checkpoint, so the checkpoint cached state can be used
+    // if desired.
+    let client = reqwest::Client::new();
+    let res = client
+        .post(format!("http://{}", &zebra_rpc_address.to_string()))
+        // Manually constructed request to avoid encoding it, for simplicity
+        .body(r#"{"jsonrpc": "2.0", "method": "getblock", "params": ["1000000", 0], "id":123 }"#)
```

Comment:
Question: This height will succeed on both the tip state and the mandatory checkpoint state. Do we only want it to succeed on the tip state?

```suggestion
        .body(r#"{"jsonrpc": "2.0", "method": "getblock", "params": ["1600000", 0], "id":123 }"#)
```

Reply:
My intention would be to work on both, just in case we ran into problems with the full sync cached state and wanted to get this done. But since we seem to be able to generated the fully sync again, I'm using the highest block from MAINNET_BLOCKS which seems to be post-mandatory checkpoint, in c9e477a62